imap_processing/mag/l0/decom_mag.py

```diff
@@ -0,0 +1,63 @@
+import logging
+from pathlib import Path
+
+from bitstring import ReadError
+from space_packet_parser import parser, xtcedef
+
+from imap_processing import imap_module_directory
+from imap_processing.ccsds.ccsds_data import CcsdsData
+from imap_processing.mag.l0.mag_l0_data import MagL0, Mode
+
+logging.basicConfig(level=logging.ERROR)
+logger = logging.getLogger(__name__)
+
+
+# TODO: write the output of this into a file
+def decom_packets(packet_file_path: str) -> list[MagL0]:
+    """Decom MAG data packets using MAG packet definition.
+
+    Parameters
+    ----------
+    packet_file_path : str
+        Path to data packet path with filename.
+
+    Returns
+    -------
+    data : list[MagL0]
+        A list of MAG L0 data classes, including both burst and normal packets. (the
+        packet type is defined in each instance of L0.)
+    """
+    # Define paths
+    xtce_document = Path(
+        f"{imap_module_directory}/mag/packet_definitions/MAG_SCI_COMBINED.xml"
+    )
+
+    packet_definition = xtcedef.XtcePacketDefinition(xtce_document)
+    mag_parser = parser.PacketParser(packet_definition)
+
+    data_list = []
+
+    with open(packet_file_path, "rb") as binary_data:
+        try:
+            mag_packets = mag_parser.generator(
+                binary_data,
+                buffer_read_size_bytes=5790778,  # Todo: what size?
+            )
+
+            for packet in mag_packets:
+                apid = packet.header["PKT_APID"].derived_value
+                if apid in (Mode.BURST, Mode.NORM):
+                    values = [
+                        item.derived_value
+                        if item.derived_value is not None
+                        else item.raw_value
+                        for item in packet.data.values()
+                    ]
+                    data_list.append(MagL0(CcsdsData(packet.header), *values))
+        except ReadError as e:
+            logger.warning(
```

Comment:
Do you want this as error? I don't think it will be logged currently because you set the level to ERROR above. I'll still maintain I think we should not be catching these and rather failing so it is loud when something breaks...

Reply:
If there is an incomplete packet, wouldn't we want to keep reading and not halt processing?

Reply:
But, you are right that it should be logged as an ERROR at least.

Reply:
I think this is a good question. Do we expect this to happen, and if so do we want to continue on and accept a missing point, or do we want to fail early and force us to go in and look at it to fix it. I'm not sure there is a right answer, and maybe different teams have different expectations?

I am sort of anticipating not getting incorrect packets down, so what I'm worried about is us updating a variable in the XTCE or something like that and then logging errors and not catching that we actually incorrectly updated the XTCE because we didn't raise. Either way is probably not going to happen is my guess...